.github/workflows/CI.yml

```diff
@@ -25,115 +38,67 @@ jobs:
-        if: matrix.rust == 'nightly'
-      - name: "gio: build nightly"
-        run: cargo build --manifest-path gio/Cargo.toml --all-features
-        if: matrix.rust == 'nightly'
-      # not nightly
-      - name: "gio: tests"
-        run: xvfb-run --auto-servernum cargo test --manifest-path gio/Cargo.toml --features "v2_66"
-        if: matrix.rust != 'nightly'
-      - name: "gio: build"
-        run: cargo build --manifest-path gio/Cargo.toml --features "v2_66"
-        if: matrix.rust != 'nightly'
-      # glib
-      # nightly
-      - name: "glib: tests nightly"
-        run: xvfb-run --auto-servernum cargo test --manifest-path glib/Cargo.toml --all-features
-        if: matrix.rust == 'nightly'
-      - name: "glib: build nightly"
-        run: cargo build --manifest-path glib/Cargo.toml --all-features
+      - name: tests
+        run: xvfb-run --auto-servernum cargo test --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "${{ matrix.conf.features }}"
+        if: matrix.rust != 'nightly'
+      - name: tests sys
+        run: xvfb-run --auto-servernum cargo test --manifest-path ${{ matrix.conf.name }}/sys/Cargo.toml --features "${{ matrix.conf.features }}"
+        if: matrix.rust != 'nightly' && matrix.conf.test_sys
+      - name: build
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "${{ matrix.conf.features }}"
+        if: matrix.rust != 'nightly'
+
+      # rustdoc-stripper
+      - name: doc add
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "dox,embed-lgpl-docs"
+        if: matrix.rust == 'nightly' && matrix.conf.name != 'glib'
+      - name: doc removal
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "purge-lgpl-docs"
+        if: matrix.rust == 'nightly' && matrix.conf.name != 'glib'
+      - name: glib doc check
+        run: cargo build --manifest-path ${{ matrix.conf.name }}/Cargo.toml --features "dox"
+        if: matrix.rust == 'nightly' && matrix.conf.name == 'glib'
+      - name: check diff
+        run: git diff -R --exit-code
         if: matrix.rust == 'nightly'
+
+      - uses: bcomnes/cleanup-xvfb@v1
+
+  build-others:
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:20.10
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+          - "1.48.0"
+    steps:
+      - run: apt-get update -y
+      - run: apt-get install -y libgtk-3-dev libglib2.0-dev libgraphene-1.0-dev git xvfb curl libcairo-gobject2 libcairo2-dev
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          override: true
+      # glib (macros)
       - name: "glib: tests glib macros nightly"
         working-directory: glib
         run: python3 clone_tests/clone_compilation_errors.py
         if: matrix.rust == 'nightly'
```

Comment:
I may be misreading the patch, but it seems to me that `python3 clone_tests/clone_compilation_errors.py` could be run unconditionally instead of duplicating it for nightly

Reply:
You're right, I think it's an artifact. Removing the condition and the duplica.